src/cluster_copier.cc

```diff
@@ -16,28 +16,14 @@ namespace grpc_labview {
     // cluster: Pointer to the cluster created by LabVIEW
     void ClusterDataCopier::CopyToCluster(const LVMessage& message, int8_t* cluster)
     {
-        std::map<std::string, int> oneof_containerToSelectedIndexMap;
-        for (auto val : message._metadata->_mappedElements)
+        for (auto indexAndValue : message._values)
```

Comment:
Note: I reversed the order of the two loops here for efficiency. I also perform a map lookup rather than linear search which effectively eliminates the second loop.